src/daemon/daemon.h

```diff
@@ -136,14 +140,32 @@ public slots:
     void release_resources(const std::string& instance);
     std::string check_instance_operational(const std::string& instance_name) const;
     std::string check_instance_exists(const std::string& instance_name) const;
-    grpc::Status create_vm(grpc::ServerContext* context, const CreateRequest* request,
-                           grpc::ServerWriter<CreateReply>* server, bool start);
+    void create_vm(const CreateRequest* request, grpc::ServerWriter<CreateReply>* server,
+                   std::promise<grpc::Status>* status_promise, bool start);
     grpc::Status reboot_vm(VirtualMachine& vm);
     grpc::Status shutdown_vm(VirtualMachine& vm, const std::chrono::milliseconds delay);
     grpc::Status cancel_vm_shutdown(const VirtualMachine& vm);
     grpc::Status cmd_vms(const std::vector<std::string>& tgts, std::function<grpc::Status(VirtualMachine&)> cmd);
     void install_sshfs(const VirtualMachine::UPtr& vm, const std::string& name);
 
+    struct AsyncOperationStatus
+    {
+        grpc::Status status;
+        std::promise<grpc::Status>* status_promise;
+    };
+
+    AsyncOperationStatus async_wait_for_ssh_for(const VirtualMachine::UPtr& vm,
+                                                std::promise<grpc::Status>* status_promise);
+    AsyncOperationStatus async_wait_for_ssh_all(const std::vector<std::string>& vms,
+                                                std::promise<grpc::Status>* status_promise);
+    template <typename Reply>
+    AsyncOperationStatus async_wait_for_ssh_and_start_mounts(grpc::ServerWriter<Reply>* server,
+                                                             const std::vector<std::string>& vms,
+                                                             std::promise<grpc::Status>* status_promise);
+    void finish_async_operation(QFuture<AsyncOperationStatus> async_future);
+    QFutureWatcher<AsyncOperationStatus>* create_future_watcher();
+    std::vector<std::unique_ptr<QFutureWatcher<AsyncOperationStatus>>> async_future_watchers;
```

Comment:
Hmm, I am not sure I like this in the beginning... I understand you wanted to group it with related methods, but the division so far was between methods and attributes, and this moves all other attributes in the object by one position. Conceptually I also think it makes more sense to have config first, then instances, etc.

Was clang-format happy with the newline here?

Reply:
Sure, I can move the attribute to the bottom of the other attributes and probably should have done that in the first place.  I'm pretty sure I ran `clang-format` on all of my commits, but I will again just to be sure.

Reply:
FYI, moved the attribute to the bottom as well as running `clang-format` on all of the changes thus far.